src/NServiceBus.Core/Persistence/InMemory/SagaPersister/InMemorySagaPersister.cs

```diff
@@ -40,8 +41,7 @@ public Task Complete(IContainSagaData sagaData, SynchronizedStorageSession sessi
                 {
                     continue;
                 }
-                var clone = (TSagaData) DeepClone(entity.SagaData);
-                entity.RecordRead(clone, version);
+                var clone = entity.Read<TSagaData>(version);
```

Comment:
Since `DeepClone` is now an implementation detail of `Read`, perhaps this variable should be named something other than `clone`?